### PR TITLE
fix: pass offsets to partitions as milliseconds instead of seconds

### PR DESCRIPTION
**Pull request by rgraber**
Kafka partitions expect time offsets to be in milliseconds instead of seconds, so multiply the timestamp by 1000 before setting it on the partition.

Tested locally against the dev cluster.

**Merge checklist:**
Check off if complete *or* not applicable:
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [x] Noted any: Concerns, dependencies, deadlines, tickets, testing instructions

### CODE REVIEW

File: edx_event_bus_kafka/internal/consumer.py [hunk 1]
```diff
@@ -146,11 +146,12 @@ def reset_offsets(consumer, partitions):
             if offset_timestamp is None:
                 return
 
-            # Get the offset from the epoch in seconds.
-            offset_timestamp_s = int(offset_timestamp.timestamp())
+            # Get the offset from the epoch (Kafka expects offsets in milliseconds for offsets_for_times, see
+            # https://kafka-python.readthedocs.io/en/master/apidoc/KafkaConsumer.html#kafka.KafkaConsumer.offsets_for_times)
```
Comment (timmc-edx) [inline edx_event_bus_kafka/internal/consumer.py:150]: Might be misleading to link to the kafka-python library's docs here, since we're not using it.

Comment (rgraber) [reply edx_event_bus_kafka/internal/consumer.py:150]: Good catch. I'll see if I can find it in the library we're actually using.

Comment (timmc-edx) [reply edx_event_bus_kafka/internal/consumer.py:150]: As far as I can tell it's undocumented, both for confluent-kafka and the underlying librdkafka. I have a PR open to add a note about it in the latter.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -14,6 +14,12 @@ Change Log
 Unreleased
 **********
 
+[3.5.1] - 2023-01-05
```
Comment (timmc-edx) [inline CHANGELOG.rst:17]:
```suggestion
[3.5.1] - 2023-01-06
```

File: edx_event_bus_kafka/internal/consumer.py (hunk 1 above)
Comment (timmc-edx) [inline edx_event_bus_kafka/internal/consumer.py:150]: Maybe something along these lines? Anyway, I'll go ahead and approve, and you can decide what makes the most sense docs-wise.

```suggestion
            # Get the offset from the epoch. Kafka expects offsets in milliseconds for offsets_for_times. Although
            # this is undocumented in the libraries we're using (confluent-kafka and librdkafa), for reference
            # see the docs for kafka-python:
            # https://kafka-python.readthedocs.io/en/master/apidoc/KafkaConsumer.html#kafka.KafkaConsumer.offsets_for_times
```

**Review by timmc-edx — APPROVED** [review]

**Pull request merged**